### DISCUSSION TITLE
Setting an Output Audio Level Cap?

### DISCUSSION THREAD

**Discussion by FOSSilizedDaemon** [pine64/OpenPineBuds#106] [General]
I have been poking around in the source code looking for where, if at all, the code for the maximum audio output level is. I was thinking it could be useful for those who want, to be able to set a maximum output level so that if they ever set their phone volume level to high the earbuds will not go past a certain volume output level. Is the ability to do this preexisting in the current source tree? I will keep looking, but haven't seen anything yet.

**Comment by Ralim** [comment]
Hello,
My first guess would to look at https://github.com/pine64/OpenPineBuds/blob/main/config%2Fopen_source%2Ftgt_hardware.c#L148 and the struct below it; they should define the output amplification
